src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrManagerBase.cs

```diff
@@ -60,35 +69,75 @@ protected void Commit(string commitMessage, Signature author)
     }
 
     /// <summary>
-    /// Parses Version.Details.xml of a given mapping and returns the list of source build dependencies (+ their mapping).
+    /// Recursively parses Version.Details.xml files of all repositories and returns the list of source build dependencies.
     /// </summary>
-    protected async Task<IList<(DependencyDetail dependency, SourceMapping mapping)>> GetDependencies(
-        SourceMapping mapping,
-        CancellationToken cancellationToken)
+    protected async Task<IEnumerable<DependencyUpdate>> GetAllDependencies(DependencyUpdate root, CancellationToken cancellationToken)
     {
-        var versionDetailsPath = _vmrInfo.GetRepoSourcesPath(mapping) / VersionFiles.VersionDetailsXml;
-        var versionDetailsContent = await File.ReadAllTextAsync(versionDetailsPath, cancellationToken);
+        var transitiveDependencies = new Dictionary<SourceMapping, DependencyUpdate>
+        {
+            { root.Mapping, root },
+        };
 
-        var dependencies = _versionDetailsParser.ParseVersionDetailsXml(versionDetailsContent, true)
-            .Where(dep => dep.SourceBuild is not null);
+        var reposToScan = new Queue<DependencyUpdate>();
+        reposToScan.Enqueue(transitiveDependencies.Values.Single());
 
-        var result = new List<(DependencyDetail, SourceMapping)>();
+        _logger.LogInformation("Finding transitive dependencies for {mapping}:{revision}..", root.Mapping.Name, root.TargetRevision);
```

Comment:
I am not sure if a `.` (dot symbol) could be part of revision.. what about to put `{mapping}:{revision}` into quotes? Cause if revision could have a dot symbol, then `{mapping}:{revision}..` might have `..` or `...` at the end.

Reply:
Valid point, however revision here is the commit SHA so we're fine